dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java

```diff
@@ -192,123 +191,128 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
-					if ( Config.getBooleanProperty("DEFAULT_FILE_TO_DEFAULT_LANGUAGE", false)
-							&& defaultLang != lang ) {
-
-						ContentletAPI contentletAPI = APILocator.getContentletAPI();
-
-						//Build the lucene query with the identifier and both languages, the default and one in session to see what we can find
-						StringBuilder query = new StringBuilder();
-						query.append("+(languageId:").append(defaultLang).append(" languageId:").append(lang).append(") ");
-						query.append("+identifier:").append(assetIdentifier).append(" +deleted:false ");
-						if ( live ) {
-							query.append("+live:true ");
-						} else {
-							query.append("+working:true ");
-						}
-
-						List<Contentlet> foundContentlets = contentletAPI.search(query.toString(), 2, -1, null, user, respectFrontendRoles);
-						if ( foundContentlets != null && !foundContentlets.isEmpty() ) {
-							//Prefer the contentlet with the session language
-							content = foundContentlets.get(0);
-							if ( content.getLanguageId() != lang && foundContentlets.size() == 2 ) {
-								content = foundContentlets.get(1);
-							}
-						} else {
-							Logger.error(this, "Content with Identifier [" + assetIdentifier + "] not found.");
-							resp.sendError(404);
-							return;
-						}
-
-					}
-					else {
-						/*
+			Contentlet content = null;
+			if(byInode) {
+			    if(isTempBinaryImage)
+			        content = contentAPI.find(assetInode, APILocator.systemUser(), respectFrontendRoles);
+			    else
+			        content = contentAPI.find(assetInode, user, respectFrontendRoles);
+			    assetIdentifier = content.getIdentifier();
+			} else {
+			    boolean live=userWebAPI.isLoggedToFrontend(req);
+			    boolean PREVIEW_MODE = false;
+			    boolean EDIT_MODE = false;
+
+			    if(session != null) {
+			        PREVIEW_MODE = ((session.getAttribute(com.dotmarketing.util.WebKeys.PREVIEW_MODE_SESSION) != null));
+			        try {
+			            EDIT_MODE = (((session.getAttribute(com.dotmarketing.util.WebKeys.EDIT_MODE_SESSION) != null)));
+			        } catch (Exception e) {
+			            Logger.error(this, "Error: Unable to determine if there's a logged user.", e);
+			        }
+			    }
+			    //GIT-4506
+			    if(WebAPILocator.getUserWebAPI().isLoggedToBackend(req)){
+			        if(!EDIT_MODE && !PREVIEW_MODE)// LIVE_MODE
+			            live = true;
+			        else
+			            live = false;
+			    }
+
+			    if (req.getSession(false) != null && req.getSession().getAttribute("tm_date")!=null) {
+			        live=true;
+			        Identifier ident=APILocator.getIdentifierAPI().find(assetIdentifier);
+			        if(UtilMethods.isSet(ident.getSysPublishDate()) || UtilMethods.isSet(ident.getSysExpireDate())) {
+			            Date fdate=new Date(Long.parseLong((String)req.getSession().getAttribute("tm_date")));
+			            if(UtilMethods.isSet(ident.getSysPublishDate()) && ident.getSysPublishDate().before(fdate))

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)